source/common/router/router.cc

```diff
@@ -1533,6 +1537,12 @@ void Filter::doRetry() {
     downstream_headers_->setEnvoyAttemptCount(attempt_count_);
   }
 
+  // TODO: Move this to an extension once support for upstream router/retry filters exists.
+  // https://github.com/envoyproxy/envoy/issues/10455
```

Comment:
This config option is forever effectively. If we think this is not the right place for this ultimately, we shouldn't add this at all. Per @PiotrSikora I do wonder whether there is a way you could hack this somehow in envoy mobile if you think we want to move this. The hack would be to grab the created API listener and using dynamic_casts get at the underlying impl and toggle a bool?

Reply:
The fact that the config option will become a part of the public API forever is a fair call out. I'm pausing on this for now while I sync up with our team internally to see if there are other options aside from upstreaming this change (including if your workaround suggestion could solve the issue). Thanks.

Reply:
One option would be to keep the `add_request_date_header` option as part of the public API (since it's a valid behavior), which would only add the `Date` request header to the original request (if absent) and retain the same value across all retries in Envoy Proxy, but which would keep updating the `Date` request header in Envoy Mobile.